common/src/auth/ServiceConnectionStringParser.cs

```diff
@@ -0,0 +1,228 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Azure.DigitalTwin.Model.Service;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Azure.Devices.Common.Authorization
+{
+    /// <summary>
+    /// The Service Connection String Parser class
+    /// </summary>
+    public class ServiceConnectionStringParser
+    {
+        protected char ValuePairDelimiter = ';';
+        protected char ValuePairSeparator = '=';
+        protected static string HostNameSeparator = ".";
+
+        protected string HostNamePropertyName = nameof(HostName);
```

Comment:
If someone renames these variables, it'll break the parser. You could choose to hard code the names instead. Or you could have unit tests that validate the property name is as expected. Personally, I'd just hard code them.

Reply:
Agreed, hard coded